### PR TITLE
Core: allow tree-shaking again for materials

### PR DESCRIPTION
**Pull request by marcofugaro**
Related issue: https://github.com/mrdoob/three.js/pull/23314#issuecomment-1021448535 (why didn't you guys ping me 😅)

**Description**

Since #23314 materials have not been tree-shakeable (meaning if you don't use `MeshPhysicalMaterial` in your scene, it still stays in the final bundle).

This is the correct way to have a `fromType()` function is to put it in the class where it's needed. `MaterialLoader` is the place, since it already handles the creation of the materials.

This PR deprecates **Material.fromType** and replaces it with **MaterialLoader.createMaterialFromType**.

```diff
- const material = Material.fromType(type)
+ const material = MaterialLoader.createMaterialFromType(type)
```

| Before this PR | After this PR |
|-------------|-------------|
| image | image |

/cc @sunag

### CODE REVIEW

File: src/loaders/MaterialLoader.js [hunk 1]
```diff
@@ -303,5 +322,31 @@ class MaterialLoader extends Loader {
...
+		MeshNormalMaterial,
+		MeshLambertMaterial,
+		MeshDepthMaterial,
+		MeshDistanceMaterial,
+		MeshBasicMaterial,
+		MeshMatcapMaterial,
+		LineDashedMaterial,
+		LineBasicMaterial,
+		Material
+	};
```
Comment (marcofugaro) [inline src/loaders/MaterialLoader.js:346]: This object can safely be in the function thanks to garbage collection.

File: examples/jsm/nodes/materials/Materials.js [hunk 2]
```diff
@@ -13,28 +12,6 @@ export {
...
-
-		return new materialLib[ type ]();
-
-	}
-
-	return fromTypeFunction.call( this, type );
-
-};
-
 NodeMaterial.fromMaterial = function ( material ) {
```
Comment (LeviPesin) [inline examples/jsm/nodes/materials/Materials.js:15]: Please also fix this function, it was using materialLib.

File: src/loaders/MaterialLoader.js [hunk 3]
```diff
@@ -6,7 +6,26 @@ import { Matrix3 } from '../math/Matrix3.js';
 import { Matrix4 } from '../math/Matrix4.js';
 import { FileLoader } from './FileLoader.js';
 import { Loader } from './Loader.js';
-import { Material } from '../materials/Material.js';
+import {
```
Comment (LeviPesin) [inline src/loaders/MaterialLoader.js:9]: Maybe use just `import * from`?

File: examples/jsm/nodes/materials/Materials.js (hunk 2 above)
Comment (marcofugaro) [reply examples/jsm/nodes/materials/Materials.js:15]: good catch!

File: examples/jsm/nodes/loaders/NodeMaterialLoader.js [hunk 4]
```diff
@@ -1,5 +1,27 @@
 import { MaterialLoader } from 'three';
```
Comment (LeviPesin) [inline examples/jsm/nodes/loaders/NodeMaterialLoader.js:1]: The file should import materials from nodes/materials/Materials.js.

File: src/loaders/MaterialLoader.js (hunk 3 above)
Comment (marcofugaro) [reply src/loaders/MaterialLoader.js:9]: I need the explicit names in the object, it's better like this

File: examples/jsm/nodes/loaders/NodeMaterialLoader.js (hunk 4 above)
Comment (marcofugaro) [reply examples/jsm/nodes/loaders/NodeMaterialLoader.js:1]: good catch!

**Comment by sunag** [comment]
@marcofugaro I tried something similar here https://github.com/mrdoob/three.js/pull/22552 I don't know what @mrdoob would think about it now.

File: examples/jsm/nodes/loaders/NodeMaterialLoader.js [hunk 5]
```diff
@@ -1,4 +1,33 @@
...
+
+MaterialLoader.createMaterialFromType = function ( type ) {
+
+	const materialLib = {
+		NodeMaterial,
+		LineBasicNodeMaterial,
+		MeshBasicNodeMaterial,
+		MeshStandardNodeMaterial,
+		PointsNodeMaterial
+	};
```
Comment (LeviPesin) [inline examples/jsm/nodes/loaders/NodeMaterialLoader.js:20]: `SpriteNodeMaterial` should also go here.

Comment (marcofugaro) [reply examples/jsm/nodes/loaders/NodeMaterialLoader.js:20]: thank you 🙏

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**